### PR TITLE
Provide arguments for extend() mode

### LINKED ISSUE #385: Provide arguments for extend() mode
Not it's impossible to use `extend()` with arrow functions because there's no arguments provided.
See https://goo.gl/DeCuG5

### PR DESCRIPTION
**Pull request by tadatuta**
Fixes #385

### CODE REVIEW

**Comment by tadatuta** [comment]
If it's ok, I'll port it to 7.x as well.

File: test/runtime-match-test.js [hunk 1]
```diff
@@ -77,4 +77,17 @@ describe('Runtime Match', function() {
       });
     }, { block: 'b', foo: 'This is' }, 'This is ContextChild');
   });
+
+  it('should pass BEMContext instance and json to extend body',
```
Comment (miripiruni) [inline test/runtime-match-test.js]: Can you move test to [test/modes-extend-test.js](https://github.com/bem/bem-xjst/blob/master/test/modes-extend-test.js)?

Comment (tadatuta) [inline test/runtime-match-test.js]: sure but aren't it consistent with other tests in this file?

**Comment by miripiruni** [comment]
👍

**Pull request merged**